### ISSUE TITLE
Exception.toString not useable in @safe

### ISSUE THREAD

**Issue by braddr** [dlang/dmd#20573]
This code (extracted from a larger project):
```
module safety;

import core.sys.posix.stdio : printf;
import std.exception;

void main() @safe
{
    try
    {
        throw new Exception("irrelevant");
    }
    catch (Exception e)
    {
        string s = e.toString();
    }
}
```

Results in:
```
safety.d(14): Error: `@safe` function `D main` cannot call `@system` function `object.Throwable.toString`
/home/braddr/sandbox/dmd.2.109.1/linux/bin64/../../src/druntime/import/object.d(2652):        `object.Throwable.toString` is declared here
```

**Comment by thewilsonator** [comment]
Yes

**Comment by braddr** [comment]
oops.. forgot that druntime was moved to the dmd repo.  I moved it out since it wasn't a dmd issue, but you're correct.

**Comment by jmdavis** [comment]
Yeah, this falls under the general set of issues that come from these functions being on `Object` - and being there from before `@safe` was even a thing to boot - though even if `Object` didn't have it, `Exception` would then have to introduce it into its own hierarchy, and that creates some of the same problems.

If we made `toString` be `@safe`, then that would break existing code. It would also force all implementations to be `@safe` or `@trusted` - which may not be unreasonable, but I don't think that we can make that change at this point. We may be able to change it with editions at some point, though attributes are always going to be a problem in a situation like this, since putting them on the base class function puts restrictions on the entire class hierarchy - which is probably reasonable in the case of `@safe` but considerably less so with most of the others.
